plugins/aggregators/quantile/quantile.go

```diff
@@ -0,0 +1,165 @@
+package quantile
+
+import (
+	"fmt"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/aggregators"
+)
+
+type Quantile struct {
+	Quantiles        []float64 `toml:"quantiles"`
+	Compression      float64   `toml:"compression"`
+	ApproximatorType string    `toml:"approximator"`
+
+	newApproximator newApproximatorFunc
+
+	cache    map[uint64]aggregate
+	suffixes []string
+}
+
+type aggregate struct {
+	name   string
+	fields map[string]approximator
+	tags   map[string]string
+}
+
+type newApproximatorFunc func(compression float64) (approximator, error)
+
+var sampleConfig = `
+  ## General Aggregator Arguments:
+  ## The period on which to flush & clear the aggregator.
+  period = "30s"
+
+  ## If true, the original metric will be dropped by the
+  ## aggregator and will not get sent to the output plugins.
+  drop_original = false
+
+	## Quantiles to output in the range [0,1]
+	# quantiles = [0.25, 0.5, 0.75]
+
+	## Type of approximation
+	## Supported are:
+	##	"t-digest" -- approximation using centroids, can cope with large number of samples
+	##  "exact R7" -- exact approximation also used by Excel or NumPy (Hyndman & Fan 1996 R7)
+	##  "exact R8" -- exact approximation (Hyndman & Fan 1996 R8)
+	## NOTE: Do not use "exact" approximations with large number of samples
+	##       to not impair performance or memory consumption!
+	# approximator = "t-digest"
+
+	## Compression for approximation (t-digest). The value needs to be
+	## greater or equal to 1.0. Smaller values will result in more
+	## performance but less accuracy.
+	# compression = 100.0
```

Comment:
Fix indentation and make this match the readme.

Reply:
Will do.